Source/arcadia/internal/compiler.clj

```diff
@@ -30,14 +31,49 @@
        (alter @#'clojure.core/*loaded-libs* into @loaded-libs'))
      nil))
 
-(defn aot [path ns-syms]
+(defn aot
   "aot ns-syms to the given path, with all dependencies"
+  [path ns-syms]
   (Arcadia.Boot/SetClojureLoadPath)
   (Arcadia.Boot/AddSourcePaths)
-  (aot-namespaces path (concat ns-syms [
-    'clojure.core
-    'clojure.core.server
-    'arcadia.internal.namespace
-    'arcadia.repl]))
+  (aot-namespaces path (concat ns-syms ['clojure.core
+                                        'clojure.core.server
+                                        'arcadia.internal.namespace
+                                        'arcadia.repl]))
   (Arcadia.Boot/SetClojureLoadPathWithDLLs)
-  (Arcadia.Boot/AddSourcePaths))
\ No newline at end of file
+  (Arcadia.Boot/AddSourcePaths))
+
+(defn- copy-infrastructure! [dlls-dir]
+  (doseq [file (System.IO.Directory/GetFiles "ArcadiaGodot/Infrastructure")
+          :when (string/ends-with? file "dll")]
+    (let [file-name (last (string/split file #"/"))]
+      (System.IO.File/Copy file (System.IO.Path/Combine dlls-dir file-name) true))))
+
+(defn- compile-project! [target namespaces]
+  (let [dlls-dir (System.IO.Path/Combine target "dlls")]
+    (System.IO.Directory/CreateDirectory dlls-dir)
+    (copy-infrastructure! dlls-dir)
+    (aot dlls-dir namespaces)))
+
+(defn- file-namespace [file]
+  (-> file (slurp :enc "utf-8") edn/read-string second symbol))
```

Comment:
This function assumes that the first statement of the file is `(ns <namespace> ,,,)`. Maybe there's a more reliable way to find a file's namespace?